rs/replay/src/ingress.rs

```diff
@@ -46,35 +54,55 @@ impl From<SignedIngress> for IngressWithPrinter {
     }
 }
 
+struct PrincipalSender(PrincipalId);
```

Comment:
Maybe leave a little comment here, why we need this. 
I think I understand, we only inject the senders principal into the agent, otherwise, this effectively behaves like the `Anonymous` identity, but documenting this would be nice.